taqueria-vscode-extension-web-ui/src/App.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import { AllEditors } from './AllEditors';
+import { MichelineEditor } from './data-editors/MichelineEditor';
+import { InteropMessageInterface } from './interopTypes';
+
+const state = {
+	interop: undefined as undefined | InteropMessageInterface,
+	onChangeCallback: (x: InteropMessageInterface) => {/* empty */},
+	logs: [] as { message: string; data?: unknown }[],
+};
+
+declare global {
+	interface Window {
+		setGlobalInteropMessageInterface: (x: InteropMessageInterface) => void;
+	}
+}
+
+// const log = (message: string, data?: unknown) => {
+// 	state.logs.push({ message, data });
+// };
+// console.log = log;
+
+const setGlobalInteropMessageInterface = (x: InteropMessageInterface) => {
+	state.interop = x;
+	state.onChangeCallback(x);
+};
+window.setGlobalInteropMessageInterface = setGlobalInteropMessageInterface;
```

Comment:
Why do we need some global stuff in React app? 😵

Reply:
This is suggested by @mweichert to avoid having to do this:
```
const w = window as unknown as Record<string, unknown>;
w.setGlobalInteropMessageInterface = setGlobalInteropMessageInterface;
```

Reply:
I mean why do we need a global function at all. I'm not talking about the typings 😛